app/models/container_image.rb

```diff
@@ -114,4 +114,46 @@ def disconnect_inv
   end
 
   alias_method :perform_metadata_sync, :sync_stashed_metadata
+
+  def refresh_openshift_information
+    return if digest.blank?
+    openshift_image = ext_management_system.connect.get_image(digest)
+  rescue KubeException => e
+    raise unless e.message.include?("not found")
+  else
+    openshift_image_data = ManageIQ::Providers::Openshift::ContainerManager::RefreshParser.parse_openshift_image_data(openshift_image)
+    EmsRefresh.save_labels_inventory(self,
+                                     openshift_image_data.delete(:labels))
+    EmsRefresh.save_docker_labels_inventory(self,
+                                            openshift_image_data.delete(:docker_labels))
```

Comment:
@cben I changed here to use "save_inventory" functions in accordance to our discussion about this. PTAL